### ISSUE TITLE
Copy only *.conf when copying experiment

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#429]
In GitLab by @pabretonniere on Aug 20, 2019, 17:00

Hi @dbeltrankyl 

I tried to copy an experiment from another user but it failed because it couldn't read a nohup file in the conf folder of the source experiment:

```
pbretonn@ bscesautosubmit01:~$ autosubmit expid -y a22k -H marenostrum4 -d "Copy of a22k to debug octal number issue in clean (issue 1032)"
The new experiment "a262" has been registered.
Copying previous experiment config directories
[ERROR] Can not create experiment: [Errno 13] Permission denied: '/esarchive/autosubmit/a22k/conf/nohup.out'
Cleaning...
Removing experiment directory...
Deleting experiment from database...
Experiment a262 deleted
```

```
pbretonn@ bscearth319:/esarchive/autosubmit/a22k/conf$ ls -lrt /esarchive/autosubmit/a22k/conf/nohup.out
-rw-------+ 1 rmahmood Earth 4201535 Jul 30 17:14 /esarchive/autosubmit/a22k/conf/nohup.out
```

I've asked the owner of the file to change the permission but I think it might be interesting to copy only the conf/*conf files when copying the experiment, other files aren't useful anyway.

Thanks!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2019, 11:22

Thanks for reporting this, indeed I think it would be better to only copy .conf we will implement it.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2019, 12:18

Autosubmit should only copy the definitive configuration files autosubmit_$expid.conf, etc. And don't read any other file the user has in conf/ folder.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 17, 2019, 09:24

Hello @pabretonniere @dbeltrankyl @mcastril 

Just to make sure we are looking at the same picture:

"autosubmit copy" should only copy the following files, if exist, from the conf directory:

* autosubmit_expid.conf
* expdef_expid.conf
* jobs_expid.conf
* platform_expid.conf
* proj_expid.conf

Please confirm.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 17, 2019, 09:36

Hi @wuruchi ,

Yes, only these files ignoring any backup.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 17, 2019, 09:39

Thank you @dbeltrankyl 

Gonna start working on it.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 17, 2019, 10:28

Hi @mcastril @dbeltrankyl 

After looking at the code, there are 2 possible ways of dealing with this problem:

1. Make a list of allowed files from conf, and copy only those in the list.

2. Handle exception generated when trying to copy a file without permissions, so it does not break the process, then the copy method will copy everything it can.

I am implementing option 1, but it is open for suggestions.

**Comment by kinow** [comment]
In GitLab by @pabretonniere on Sep 17, 2019, 10:38

I wouldn't overcomplicate this, as this a minor issue that doesn't happen very often.

So option 1. seems good with 1 and the list provided by @dbeltrankyl .

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 17, 2019, 10:40

mentioned in commit 42f5cc8e563ca94a0b44aa1d16e21ac90ee12e2d

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 17, 2019, 10:43

Hello team,

A first merge request has been done with the approach explained before. The list has the form:

` conf_copy_filter = ["autosubmit_" + str(copy_id) + ".conf",
                                        "expdef_" + str(copy_id) + ".conf",
                                        "jobs_" + str(copy_id) + ".conf",
                                        "platforms_" + str(copy_id) + ".conf",
                                        "proj_" + str(copy_id) + ".conf"]`

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 17, 2019, 11:47

I also think `1` is the best approach. As we discussed in our meeting a copy of an experiment should only copy the files that Autosubmit uses, and avoid copying backup files and temporal files that only increase al the garbage we have spread through /esarchive/autosubmit

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2019, 15:09

closed via merge request !145

**Comment by kinow** [comment]
In GitLab by @wuruchi on Sep 18, 2019, 15:09

closed via commit 42f5cc8e563ca94a0b44aa1d16e21ac90ee12e2d

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2019, 15:09

mentioned in commit d5b590913e5f7c959ca9451db7be033a2aae41dc
